packages/kbn-dev-utils/src/worker/index.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import ChildProcess from 'child_process';
+import { Readable } from 'stream';
+import * as Rx from 'rxjs';
+
+import { REPO_ROOT } from '@kbn/repo-info';
+import { SomeDevLog } from '@kbn/some-dev-log';
+import { observeLines } from '@kbn/stdio-dev-helpers';
+
+// import type { Result } from './kibana_worker';
+
+interface StartTSWorkerArgs {
+  log: SomeDevLog;
+  /** Path to worker source */
+  src: string;
+  /** Defaults to repo root */
+  cwd?: string;
+}
+
+/**
+ * Provide a TS file as the src of a NodeJS Worker with some built-in handling
+ * of std streams and debugging.
+ */
+export function startTSWorker<Message>({ log, src, cwd = REPO_ROOT }: StartTSWorkerArgs) {
```

Comment:
New utility that does all the RX piping and starting of a node js worker for us.